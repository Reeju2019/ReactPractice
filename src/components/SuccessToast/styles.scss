.succ-card-part-1 {
  height: 50%;
  width: 100%;
  border-radius: 15px 15px 0 0;
  background-color: #8bc349;
  padding: 30px;
  margin: 0;
}

.succ-card-part-2 {
  height: 50%;
  width: 100%;
  background-color: #f1f1f1;
  border-bottom-left-radius: 15px;
  border-bottom-right-radius: 15px;
  padding: 40px;
  margin: 0;
}

.succ-img {
  width: 75%;
  border-radius: 50%;
  margin: 0;
}

.succ-status {
  font-size: 1rem;
  font-weight: 700;
  opacity: 0.8;
}

.succ-msg {
  font-size: 0.8rem;
  font-weight: 400;
  opacity: 0.7;
}

.vl {
  border-left: 2px solid #f1f1f1;
  height: 60px;
  padding: 0;
  margin: 0;
  width: 3%;
}

.succ-img-bg {
  background-color: #8bc34a;
  padding: 0;
  height: 72%;
  margin: 0;
  left: 4%;
  width: 20%;
  position: absolute;
}

.btn-close {
  position: absolute;
  right: 10%;
  top: 41%;
  &:focus {
    outline: 0;
    box-shadow: none;
  }
}

.toast-container {
  z-index: 5000;
}
