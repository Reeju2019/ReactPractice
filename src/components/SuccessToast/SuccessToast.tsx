import React from "react";

import {
  ToastContainer,
  Toast,
  Row,
  Col,
  Button,
  Image,
} from "react-bootstrap";
import { AiFillCloseCircle } from "react-icons/ai";

import "./styles.scss";

/**
 * @description Props of SuccessToast Component
 * @memberof SuccessToast
 * @property {string} logo Logo URL
 * @property {string} color Set the background color
 * @property {string} status Set the status
 * @property {string} msg Set the message
 * @property {any} position Set the position of the modal
 * @property {number} timeout Set the timeout
 */
/* eslint-disable */
interface IProps {
  logo: string;
  color: string;
  status: string;
  msg: string;
  position: any;
  timeout: number;
}

/**
 * @description States of SuccessToast Component
 * @memberof SuccessToast
 * @property {boolean} show - show
 */
/* eslint-disable */
interface IState {
  show: boolean;
}

/**
 * @component
 * This component was used in App component
 * @description SuccessToast component
 */
class SuccessToast extends React.Component<IProps, IState> {
  state: IState = {
    show: true,
  };

  /**
   * @constructor
   * @description Constructor
   * @param {IProps} props Props
   * @returns void
   */
  constructor(props: IProps) {
    super(props);
    this.closeHandler = this.closeHandler.bind(this);
  }

  /**
   * @description Close modal
   * @property {function} closeHandler Close modal
   * @memberof SuccessToast
   * @return {void}
   */
  closeHandler() {
    this.setState({ show: false });
  }

  render() {
    let tmp = this.props.position;
    return (
      <>
        <Row className="toast-container">
          <ToastContainer className="p-3" position={tmp}>
            <Toast
              show={this.state.show}
              onClose={this.closeHandler}
              delay={this.props.timeout}
              autohide
            >
              <Toast.Header>
                <Row className="text-start d-flex justify-content-center align-items-center m-0 p-0">
                  <Col
                    md={2}
                    className="d-flex justify-content-center align-items-center succ-img-bg"
                  >
                    <Image className="succ-img" src={this.props.logo}></Image>
                  </Col>
                  <Col md={9} className="">
                    <Row className="succ-status text-start ms-4">
                      {this.props.status}
                    </Row>
                    <Row className="succ-msg text-start ms-4">
                      {this.props.msg}
                    </Row>
                  </Col>
                  <Col md={1} className="vl ms-1"></Col>
                </Row>
              </Toast.Header>
            </Toast>
          </ToastContainer>
        </Row>
      </>
    );
  }
}

export { SuccessToast };
export default SuccessToast;
