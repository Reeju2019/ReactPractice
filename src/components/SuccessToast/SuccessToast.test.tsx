import { shallow } from "enzyme";
import ReactDOM from "react-dom";
import SuccessToast from "./SuccessToast";

const data = {
  status: "Login Successful!",
  msg: "Congratulations! Your account has been successfully created",
  position: "top-start",
  color: "#8bc349",
  timeout: 4000,
};

/**
 * Unit testing for Success Toast
 */
describe("Success toast Testing", () => {
  it("render without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <SuccessToast
        logo="/assets/logo/reg-suc.png"
        color={data.color}
        status={data.status}
        msg={data.msg}
        position={data.position}
        timeout={data.timeout}
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it("should render initial layout", () => {
    const component = shallow(
      <SuccessToast
        logo="/assets/logo/reg-suc.png"
        color={data.color}
        status={data.status}
        msg={data.msg}
        position={data.position}
        timeout={data.timeout}
      />
    );
    expect(component.getElements()).toMatchSnapshot();
  });
});
