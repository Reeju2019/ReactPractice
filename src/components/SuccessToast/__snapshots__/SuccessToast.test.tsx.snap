// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success toast Testing should render initial layout 1`] = `
Array [
  <React.Fragment>
    <Row
      className="toast-container"
    >
      <ToastContainer
        className="p-3"
        position="top-start"
      >
        <Toast
          autohide={true}
          delay={4000}
          onClose={[Function]}
          show={true}
        >
          <ToastHeader
            closeButton={true}
            closeLabel="Close"
          >
            <Row
              className="text-start d-flex justify-content-center align-items-center m-0 p-0"
            >
              <Col
                className="d-flex justify-content-center align-items-center succ-img-bg"
                md={2}
              >
                <Image
                  className="succ-img"
                  fluid={false}
                  rounded={false}
                  roundedCircle={false}
                  src="/assets/logo/reg-suc.png"
                  thumbnail={false}
                />
              </Col>
              <Col
                className=""
                md={9}
              >
                <Row
                  className="succ-status text-start ms-4"
                >
                  Login Successful!
                </Row>
                <Row
                  className="succ-msg text-start ms-4"
                >
                  Congratulations! Your account has been successfully created
                </Row>
              </Col>
              <Col
                className="vl ms-1"
                md={1}
              />
            </Row>
          </ToastHeader>
        </Toast>
      </ToastContainer>
    </Row>
  </React.Fragment>,
]
`;
